include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,775 @@
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct_other(const _Extents &__extents, _RankType __padding_stride_idx, const _OtherMapping &__other_mapping)
+  {
+    return __subs_type::__construct(__extents, extents<typename _Extents::index_type, _ActualPaddingStride>(__other_mapping.stride(__padding_stride_idx)));
+  }
+};
+
+template<size_t _ExtentToPadIdx, class _Extents, size_t _ActualPaddingStride>
+struct __inner_extents_type<_ExtentToPadIdx, _Extents, _ActualPaddingStride, enable_if_t<_Extents::rank() <= size_t(1)>>
+{
+  using __type = _Extents;
+
+  template<size_t _PaddingStride>
+  MDSPAN_INLINE_FUNCTION
+  static constexpr const _Extents &
+  __construct(const _Extents &__extents)
+  {
+    return __extents;
+  }
+
+  template <size_t _PaddingStride, class _Size>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct(const _Extents &__extents, _Size)
+  {
+    return __extents;
+  }
+
+  template <size_t _PaddingStride, class _RankType, class _OtherMapping>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct_other(const _Extents &__extents, _RankType, const _OtherMapping &)
+  {
+    return __extents;
+  }
+};
+
+template <size_t _ExtentToPadIdx, class _Extents, typename _Enabled = void>
+struct __unpadded_extent_type_impl
+{
+  using __type = extents<typename _Extents::index_type, _Extents::static_extent(_ExtentToPadIdx)>;
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct(const _Extents &__extents)
+  {
+    return __type(__extents.extent(_ExtentToPadIdx));
+  }
+};
+
+template <size_t _ExtentToPadIdx, class _Extents>
+struct __unpadded_extent_type_impl<_ExtentToPadIdx, _Extents, enable_if_t<(_Extents::rank() == 0)>>
+{
+  using __type = extents<typename _Extents::index_type>;
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct([[maybe_unused]] const _Extents &__extents)
+  {
+    return __type{};
+  }
+};
+}
+
+template <size_t padding_stride>
+template <class Extents>
+class layout_left_padded<padding_stride>::mapping {
+public:
+  using extents_type = Extents;
+  using index_type = typename extents_type::index_type;
+  using size_type = typename extents_type::size_type;
+  using rank_type = typename extents_type::rank_type;
+  using layout_type = layout_left_padded<padding_stride>;
+
+#ifndef MDSPAN_INTERNAL_TEST
+private:
+#endif // MDSPAN_INTERNAL_TEST
+
+  static constexpr rank_type __padding_stride_idx = 1;
+  static constexpr rank_type __extent_to_pad_idx = 0;
+
+  static_assert((padding_stride != 0)
+                || (extents_type::static_extent(__extent_to_pad_idx) == 0)
+                || (extents_type::static_extent(__extent_to_pad_idx) == dynamic_extent),
+                "if padding stride is 0, static_extent(0) must also be 0 or dynamic_extent");
+
+  static constexpr size_t __actual_padding_stride = detail::__get_actual_static_padding_stride<extents_type, padding_stride, __extent_to_pad_idx>();
```

Comment:
I think it would be useful because we kind of "hide" the actual stride which could be interesting for users to query for performance reasons. Maybe this should be a question we ask LEWG?

Reply:
@nmm0 P2642 hasn't finished LEWG review yet, so we can add whatever we like.  I've been tempted to rename the template parameter, e.g., `padding_or_alignment`, and reserve `padding_stride` for what we're calling _`actual-padding-stride`_.  Right now, the template parameter's name isn't quite accurate.

Reply:
That makes sense to me. Especially if we make this public it's a bit odd to have "padding_stride" and "actual_padding_stride", it's not clear which one is the one the user provided